project-chip/connectedhomeip
# ESP32 - fix parallel build configuration (sdkconfig) and add ipv6 variant

#### Problem
`build_examples.py` for esp32 were not working in parallel mode:
   - sdkconfig is always forced into the 'project root' so it is impossible to consistently rebuild just with ninja as the latest config will be taken

IPV6-only is not supported as a config flag for ESP32 (does not move through makefile configuration)

Fixes #9673

#### Change overview
make esp32 builds use `idf.py` for building as well, to regenerate sdkconfig if needed
ensure skdconfig is cleared before every generation/build
add ipv4-disable support in CMakeLists     
add ipv6-only build variant for esp32 in `build_examples.py`
removed `cwd` argument from build runners: all commands are assumed to be executed in the project root.

#### Testing
Manually:
   - add assert for IPV4 being disabled when expected (i.e. for ipv6only variant)
   - add logs in CMakeLists to validate that expected BT/RPC/IPV4 settings are actually taking effect

Automated: CI will validate this.
